multi-int-store/src/test/java/com/connexta/multiintstore/S3Tests.java

```diff
@@ -0,0 +1,385 @@
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.MultipartBodyBuilder;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = WebEnvironment.DEFINED_PORT)
+@DirtiesContext
+@MockBean(SolrClient.class)
+public class S3Tests {
+
+  private static final String MINIO_ADMIN_ACCESS_KEY = "admin";
+  private static final String MINIO_ADMIN_SECRET_KEY = "12345678";
+  private static final int MINIO_PORT = 9000;
+
+  @ClassRule
+  public static final GenericContainer minioContainer =
+      new GenericContainer("minio/minio:RELEASE.2019-07-10T00-34-56Z")
+          .withEnv("MINIO_ACCESS_KEY", MINIO_ADMIN_ACCESS_KEY)
+          .withEnv("MINIO_SECRET_KEY", MINIO_ADMIN_SECRET_KEY)
+          .withExposedPorts(MINIO_PORT)
+          .withCommand("server --compat /data")
+          .waitingFor(
+              new HttpWaitStrategy()
+                  .forPath("/minio/health/ready")
+                  .withStartupTimeout(Duration.ofSeconds(30)));
+
+  @TestConfiguration
+  static class Config {
+
+    @Bean
+    public AmazonS3Configuration testAmazonS3Configuration() {
+      return new AmazonS3Configuration(
+          String.format(
+              "http://%s:%d",
+              minioContainer.getContainerIpAddress(), minioContainer.getMappedPort(MINIO_PORT)),
+          "local",
+          MINIO_ADMIN_ACCESS_KEY,
+          MINIO_ADMIN_SECRET_KEY);
+    }
+  }
+
+  @Autowired private TestRestTemplate restTemplate;
+  @Autowired private AmazonS3 amazonS3;
+
+  @Value("${endpointUrl.retrieve}")
+  private String endpointUrlRetrieve;
+
+  @Value("${aws.s3.bucket.quarantine}")
+  private String s3Bucket;
+
+  @Before
+  public void before() {
+    amazonS3.createBucket(s3Bucket);
+  }
+
+  @After
+  public void after() {
+    amazonS3.listObjects(s3Bucket).getObjectSummaries().stream()
+        .map(S3ObjectSummary::getKey)
+        .forEach(key -> amazonS3.deleteObject(s3Bucket, key));
+    amazonS3.deleteBucket(s3Bucket);
+  }
+
+  @Test
+  public void testContextLoads() {}
+
+  @Test
+  public void testStoreProductWhenS3IsEmpty() throws Exception {
+    // given
+    final InputStream inputStream =
+        IOUtils.toInputStream(
+            "All the color had been leached from Winterfell until only grey and white remained",
+            StandardCharsets.UTF_8);
+    final long fileSize = (long) inputStream.available();
+    final String fileName = "test_file_name.txt";
+    final MultipartBodyBuilder multipartBodyBuilder = new MultipartBodyBuilder();
```

Comment:
We should be using a `MulitValueMap<String, Object>` to create the multipart request. https://github.com/spring-projects/spring-boot/issues/12579#issuecomment-375305699. With this change we won't need to pull in ReactiveStreams as a dependency.

The next Spring release will update the docs to make this recommendation.